parcels/particle.py

```diff
@@ -117,6 +117,7 @@ class ScipyParticle(_Particle):
     lon = Variable('lon', dtype=np.float32)
     lat = Variable('lat', dtype=np.float32)
     time = Variable('time', dtype=np.float64)
+    trajectory = Variable('trajectory', dtype=np.int32)
```

Comment:
OK, so this seems to be a deep bug within the JIT cptr. If I simply rename `trajectory` to `id`, the JIT goes all wonky, including segmentation faults on some tests..

One solution is to call the variable `xid`. My hypothesis now is that JIT is fine as long as this variable is copied *after* the `time` variable (i.e. has to be lexicographically later, like `trajectory` or `xid`). 

Something about alignment of the `cptr`??